contracts/contracts/Job.sol

```diff
@@ -335,6 +357,59 @@ contract Job is IJob, Ownable {
         emit JobDisputeResolved(jobId, States.FinalDisputeResolvedWithSplit);
     }
 
+    // Used to prevent illegal activity
+    function reportJob(uint256 jobId, string memory metadata) external {
+        JobData memory job = jobs[jobId];
+
+        // TODO: Should we also allow Disputed state ?
+        require(job.state == States.Available || job.state == States.Started, "Method not available for job state");
+
+        jobs[jobId].isReported = true;
+        reports[jobId].reporter = msg.sender;
+        reports[jobId].metadata = metadata;
+
+        receiveFunds(REPORT_TOKEN, msg.sender, REPORT_DEPOSIT);
+
+        emit JobReported(jobId, msg.sender, metadata);
+    }
+
+    function declineReport(uint256 jobId, string memory reason) external onlyResolver {
+        require(jobs[jobId].isReported, "Method not available for job state");
+
+        address reporter = reports[jobId].reporter;
+
+        delete jobs[jobId].isReported;
+        delete reports[jobId];
```

Comment:
I think it is ok to keep the reports.  There is no cost to keeping them and I think it is beneficial for transparency.